var/spack/repos/builtin/packages/sim-model/package.py

```diff
@@ -100,34 +111,40 @@ def install(self, spec, prefix, install_src=True):
         share/ <- neuron & coreneuron mod.c's (modc and modc_core)
         """
         mkdirp(prefix.bin)
-        mkdirp(prefix.lib.mod, prefix.lib.hoc)
+        mkdirp(prefix.lib)
         mkdirp(prefix.share.modc)
 
         self._install_binaries()
 
-        if spec.satisfies('+coreneuron'):
-            install = which('nrnivmech_install.sh', path=".")
-            install(prefix)
-
         if install_src:
             self._install_src(prefix)
 
-    def _install_binaries(self, lib_suffix=''):
+    def _install_binaries(self, mech_name=None):
         # Install special
+        mech_name = mech_name or self.mech_name
         arch = os.path.basename(self.neuron_archdir)
         prefix = self.prefix
+
+        if self.spec.satisfies('+coreneuron'):
+            with working_dir('build_' + mech_name):
+                if self.spec.satisfies('^coreneuron@0.14:0.16.99'):
+                    which('nrnivmech_install.sh', path=".")(prefix)
+                elif self.spec.satisfies('^coreneuron@0.17:'):
+                    which('nrnivmodl-core')("-d", prefix)  # Set dest to install
+
+        # Install special
         shutil.copy(join_path(arch, 'special'), prefix.bin)
 
         if self.spec.satisfies('^neuron~binary'):
             # Install libnrnmech - might have several links.
-            for f in find(arch + "/.libs", 'libnrnmech*.so*', recursive=False):
+            for f in find(arch + '/.libs', 'libnrnmech*.so*', recursive=False):
```

Comment:
' -> " makes the diff harder to read…

Reply:
attempt at standardizing... but work is still not complete, so sorry it came mixed here, took the chance since touched a lot this file anyway.